client/express-checkout/utils/check-payment-method-availability.js

```diff
@@ -29,7 +30,9 @@ export const checkPaymentMethodIsAvailable = memoize(
 				<Elements
 					stripe={ api.loadStripe() }
 					options={ {
-						mode: 'payment',
+						mode: getExpressCheckoutData( 'has_free_trial' )
+							? 'subscription'
+							: 'payment',
 						...( isManualPaymentMethodCreation( paymentMethod ) && {
```

Comment:
```suggestion
						...( isManualPaymentMethodCreation( paymentMethod, hasFreeTrial ) && {
```

Shouldn't we send the `hasFreeTrial` param here to set `paymentMethodCreation` as `manual` for free trial subs?

Reply:
Strangely, everything was working fine without the missing parameters like this one. We may not need some of those similar code blocks. Anyway, to keep it consistent, I have added it to all calls of this method in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4557/commits/80d57e3b235f40c2970e0998c16b72559d844e02